network/p2p/dns/resolver.go

```diff
@@ -105,7 +92,7 @@ func (r *Resolver) lookupIPAddr(ctx context.Context, domain string) ([]net.IPAdd
 	}
 
 	if !fresh && r.shouldResolveIP(domain) {
-		r.unit.Launch(func() {
```

Comment:
What is the reason behind replacing `unit` with a `goroutine`? existence of `unit` is crucial to perform a bottom-up graceful shutting down of the node's component. Replacing it with a simple `goroutine` makes an incomplete graceful shutting down. This is specifically important when it comes to the networking layer, since it stops dispatching any message in the network the moment stopping the node triggers in: 
https://github.com/onflow/flow-go/blob/bf146a8e63d2626c3d2533a4f0e95284959c2030/engine/unit.go#L46-L57

🚩  more importantly we are planning to refactor this DNS resolver to a [worker-based](https://github.com/dapperlabs/flow-go/issues/5868) one, which will be entirely relying on a lifecycle management component like `unit` to startup and shutdown the workers.

Reply:
This is because currently there's no need for `unit` in the resolver, because it doesn't actually need to be a component with `Ready` and `Done` methods as far as I can tell. Whatever component owns the resolver is responsible for stopping calls to the resolver as soon as it's stopped. If you're refactoring to a worker-based design, I think it would be much better to use `ComponentManager` instead of `unit`.

Reply:
Then please refactor it to `ComponentManager` instead of bare `goroutines`, thanks!

Reply:
f38a0a4